marklogic-data-hub/src/test/java/com/marklogic/hub/step/impl/IngestWithSourceNameAndSourceTypeTest.java

```diff
@@ -0,0 +1,68 @@
+package com.marklogic.hub.step.impl;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.marklogic.client.DatabaseClient;
+import com.marklogic.client.datamovement.WriteBatcher;
+import com.marklogic.client.document.ServerTransform;
+import com.marklogic.client.io.DocumentMetadataHandle;
+import com.marklogic.client.io.Format;
+import com.marklogic.client.io.JacksonHandle;
+import com.marklogic.client.io.StringHandle;
+import com.marklogic.hub.AbstractHubCoreTest;
+import org.junit.jupiter.api.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class IngestWithSourceNameAndSourceTypeTest extends AbstractHubCoreTest {
+
+    @Test
+    void runDefaultIngestionWithSourceNameAndSourceType(){
+        ServerTransform serverTransform = new ServerTransform("mlRunIngest");
+        Map<String, Object> options = new HashMap<>();
+        options.put("sourceName", "user-defined-source-name");
+        options.put("sourceType", "user-defined-source-type");
+        String optionString = jsonToString(options);
+        serverTransform.addParameter("options", optionString);
+
+        DatabaseClient stagingClient = getHubClient().getStagingClient();
+        WriteBatcher writeBatcher = stagingClient.newDataMovementManager().newWriteBatcher()
```

Comment:
We still want to run the ingestion step here via FlowRunner. The reason that @srinathgit  uses WriteBatcher directly in RunDefaultIngestionTest is so that he can test the mlRunIngest transform in isolation, without running a step. 

But for this test, we need to run a step. You can use the all-steps-referenced test project - it has some test data in it along with an ingestion step. See AllStepsReferencedTest as an example.